### ISSUE TITLE
[BUG]: AEBNiii Agent creates collections despite settings for no collections

### ISSUE THREAD

**Issue by fourstix** [CodyBerenson/PGMA-Modernized#203]
# Mandatory # 

## Describe the bug: ##  The AEBNiii agent is creating collections even though the settings for the agent have all the option boxes for collections unselected. 

(This seems to be the same issue that was fixed with the GEVI agent in the last code update)

## Index Site URL(s) Attempting to Match: ##
https://gay.aebn.com/gay/movies/157987/vintage-muscle-jocks  (matched successfully, but collections created)

## Log Attached: ##
[com.plexapp.agents.AEBNiii.log](https://github.com/CodyBerenson/PGMA-Modernized/files/10095233/com.plexapp.agents.AEBNiii.log)

# Optional #
The text description indicates a correct match with the AEBNiii agent:

:: Film on IAFD 👎 :: ✅ / ❌ Actor on Cast List? :: AEBNiii ::
Lavender Lounge Studios is pleased to announce the release of “Vintage Muscle Jocks”, a compilation of rare and obscure male erotica spanning the late 1950s through the late 1970s. 

## Screenshot(s) with all nudity redacted: ##

Here are the AEBNiii agent settings:
![image](https://user-images.githubusercontent.com/5274281/204060984-e5dac191-e077-4dd3-b6a7-c2b7ef323006.png)
![image](https://user-images.githubusercontent.com/5274281/204061282-0b4124f1-a33c-4da9-bd86-ef144a7332c4.png)
![image](https://user-images.githubusercontent.com/5274281/204061099-0a88aeae-c403-4051-a391-852cb9acefa1.png)

## Additional Context: ##
Matching file (Lavender Lounge) - Vintage Muscle Jocks.mp4  will create three collections: AEBNiii, Not On-IAFD and Not Stacked.  

![image](https://user-images.githubusercontent.com/5274281/204060964-fb57a8f9-e762-46df-93dc-d495378ac1d9.png)

## Desktop (please complete the following information): ##
 - OS: QNAP QTS 5.0.1.2194
 - Browser : Firefox
 - Version: 107.0 (64-bit)

**Comment by kellys297** [comment]
I'm having the same issue but noticed it with Fagalicious and Waybig; collections being created even though it's disabled in the settings. Since the last update, I've also had to manually enter the Plex token for Fagalicious (and GEVI) to get it to pull any data. I'll follow this thread with great interest.

**Comment by JPH71** [comment]
Sorted! Mark as solved

**Comment by CodyBerenson** [comment]
@fourstix I'm going to close this issue, as I'm unable to get the agents to create collections when the preferences are not checked.  Do be sure to 1) use the latest code, 2) ensure each agent's preferences are set to what you want, 3) kill PMS, and 4) restart PMS.  At least on my system, updates I make to the preferences don't take effect until PMS is killed and restarted.

Cheers, and happy holidays!

**Comment by fourstix** [comment]
Hi Cody, I am using the latest code and it is still happening for me.  I have done all three steps you listed and I still get the collections being created.

**Comment by fourstix** [comment]
Please re-open this issue, maybe it's dependent on the OS or something.

**Comment by CodyBerenson** [comment]
@fourstix @JPH71  Let me know how you want to proceed...work offline with each other?

**Comment by fourstix** [comment]
I'm fine with working offline, just curious are you using Windows or Linux?  Can you post the configuration screenshots that work for you? (Maybe I'm missing something or it might help to identify one configuration settings that work vs one that doesn't.)

**Comment by CodyBerenson** [comment]
For the dirty stuff, I'm using a Windows PMS.   Here's my AEBN that creates no collections.   I recommend you create a testing library.  Only copy one or two titles to it.  Double check your agent settings, kill PMS, restart PMS, and if it still creates the collections please upload your log file for @JPH71 to look at.  Perhaps there are Linux environmental differences that are mucking things up.  

![image](https://user-images.githubusercontent.com/40373801/208718280-c17b05d8-6530-47e8-8c0e-ec753d060588.png)
![image](https://user-images.githubusercontent.com/40373801/208718678-45cba5ae-4415-4229-978d-b11b3bc09af2.png)

**Comment by fourstix** [comment]
I also have "Use Back Art" and "Place Cast Legend" checked, along with the Plex Token value set.  I'm going to try setting "Clear Previously set meta-data" first...  then do then start and stop Plex and try again.

**Comment by CodyBerenson** [comment]
@fourstix the only way to get rid of collections in the collections tab is to physically delete them yourselves.  This is a feature of Plex.  Our agent may clear tags out of any given collection, but it will not delete the collection.  Hope that makes sense.  This is why I suggested starting with a clean library.

**Comment by fourstix** [comment]
I did delete the collections from the library before I did the match.  It still created collections when I matched the file (All Male Studio) - Gay Italiano.mp4 in AEBNiii.  

![image](https://user-images.githubusercontent.com/5274281/208724083-13046dc6-0571-483a-b825-ea2e818f1f18.png)
![image](https://user-images.githubusercontent.com/5274281/208724206-62c98266-c63e-43b8-b2a9-75e5103e1d75.png)
![image](https://user-images.githubusercontent.com/5274281/208724353-f5810c22-7a70-4fee-9a30-e42cf3290733.png)
![image](https://user-images.githubusercontent.com/5274281/208724703-ba29c80a-9cc2-4f15-9d56-59a48877a9ad.png)

I'll try creating a new library and adding that single file to it.

**Comment by fourstix** [comment]
Same behavior.  The new library Test has 3 collections created.

![image](https://user-images.githubusercontent.com/5274281/208726430-5984e27f-bcbc-45d1-aecf-ced2b46c049e.png)

**Comment by CodyBerenson** [comment]
Please upload your AEBN log for Jason to review and diagnose.

**Comment by fourstix** [comment]
Here it is:
[com.plexapp.agents.AEBNiii.log](https://github.com/CodyBerenson/PGMA-Modernized/files/10270704/com.plexapp.agents.AEBNiii.log)

**Comment by CodyBerenson** [comment]
There's the problem.  IThe agent has been renamed, and AEBNiii has been decommissioned.  

Before you copy new sets of agents to your plugins folder, you should DELETE all previous agents and paste in brand new agents.

kill your PMS.  take a backup of your agents folder and then delete all of the PGMA agents and folders.  There are 24 folders.  Ensure that the plugins folder is empty.  paste the brand new agents there.  When you relaunch PMS, you should no longer see AEBNiii, and just AEBN.

**Comment by CodyBerenson** [comment]
![image](https://user-images.githubusercontent.com/40373801/208728460-30e04834-3b05-48ca-b300-f66e649af930.png)

**Comment by fourstix** [comment]
Good catch!  No need to apologize.  That makes perfect sense.

**Comment by fourstix** [comment]
Oh, wow. Now it is creating *all* the collections! There's even one for each actor.

![image](https://user-images.githubusercontent.com/5274281/208731936-ce7ff783-5433-4d20-adc0-6f2cf0bbe9df.png)
![image](https://user-images.githubusercontent.com/5274281/208732079-1d0f5db2-e2f5-4bf8-9176-fabc42eeca1a.png)

**Comment by fourstix** [comment]
Here's the log file:
[com.plexapp.agents.AEBN.log](https://github.com/CodyBerenson/PGMA-Modernized/files/10270909/com.plexapp.agents.AEBN.log)

**Comment by fourstix** [comment]
And here are the settings:
![image](https://user-images.githubusercontent.com/5274281/208732942-e22d7305-95a8-4314-8e39-5a4bc4cd2380.png)
![image](https://user-images.githubusercontent.com/5274281/208733070-4a2a52a6-c37c-46ae-9301-9800c6c3c1ff.png)

**Comment by JPH71** [comment]
The preferences the agents use are a pain in the arse.
You have to open each agent's settings then, tick then untick then click
save.
As you have replaced all then agents, you need to do this to get the
settings recognised - thats why in the log files there is a section which
states the settings default and what they are set as currently.
This part of the log shows up just after you restart the server.

**Comment by JPH71** [comment]
From the last log file you sent, this initial preference set up ran. and
stopped.
And the settings are as you want - no collections.
1. go into the library
2. Delete all the collections
3. Access the film's tags, by clicking on the lower left icon when you
hover over the title.
5. Delete the collection info from in here
6. Make sure that none of the fields are locked (showing a yellow padlock
to the left of the field)
7. Save
8 Refresh metadata......

You can do this for the whole library, by selecting all the films, then
IMPORTANT: locking the fields, then RE-SELECT and unlocking all the fields.
After which you can refresh the metadata....

**Comment by JPH71** [comment]
Ooh - just seen this
AEBNiii is now AEBN
GayDVDEmpire is now GayEmpire...

Stop Plex - delete those bundles - copy the new ones in, save the
preferences, re-start plex...

**Comment by fourstix** [comment]
OK.  Thanks for pointing that out.  It makes sense.

I stopped Plex and deleted those bundles.  I also deleted nymMedia.bundle which was hanging around from an older release.  

I then started Plex and went through and update all the settings for each agent I use. Then I went and checked the log file for each one.   Believe it or not, I had to redo a couple of agent settings to get all the collections unselected.  After doing all that.  

I retested, and now it is working as expected and *not* creating the collections when it matches. Whew!  

Thank you for all your patience and help.  We can close this issue now.
